### DISCUSSION TITLE
No commit found for SHA

### DISCUSSION THREAD

**Discussion by CraigChamberlain** [community/community#24521] [Apps, API and Webhooks]
<p>I’m getting an error when I try and restore a paket.  Please see item at bottom.  I can’t see anywhere that that a feature needs enabling.</p>
<p>Problem is reproducible in browser.  The following gets an error response<br>
<a href="https://api.github.com/repos/CraigChamberlain/moon-data" class="onebox" target="_blank" rel="nofollow noopener">https://api.github.com/repos/CraigChamberlain/moon-data</a></p>
<p>Where as this example servers a full json object.<br>
<a href="https://api.github.com/repos/forki/FsUnit" class="onebox" target="_blank" rel="nofollow noopener">https://api.github.com/repos/forki/FsUnit</a></p>
<pre><code class="lang-auto">Could not find hash for https://api.github.com/repos/CraigChamberlain/moon-data/commits/
RequestFailedException: Request to 'https://api.github.com/repos/CraigChamberlain/moon-data/commits/' failed with: 422 UnprocessableEntity — {"message":"No commit found for SHA: ","documentation_url":"https://developer.github.com/v3/repos/commits/#get-a-single-commit"}
</code></pre>

**Comment by francisfuzz** [comment, marked as answer]
<p>:wave: <a class="mention" href="/craigchamberlain">@CraigChamberlain</a>, welcome to the GitHub Support Community! :sunglasses:</p>
<p>Thanks for asking about this.</p>
<p>The following URLs are structured to <a href="https://developer.github.com/v3/repos/#get-a-repository">request a single repository</a> based on a repository owner and its name:</p>
<ul>
<li><a href="https://api.github.com/repos/CraigChamberlain/moon-data">https://api.github.com/repos/CraigChamberlain/moon-data</a></li>
<li><a href="https://api.github.com/repos/forki/FsUnit">https://api.github.com/repos/forki/FsUnit</a></li>
</ul>
<p>Sending a <a href="https://tools.ietf.org/html/rfc7231#section-4.3.1"><code>GET</code> request</a> to either of these URLs should return a <a href="https://httpstatuses.com/200"><code>200 OK</code></a> because they’re both public repositories.</p>
<p>:mag: Let’s look at this line together:</p>
<pre><code class="lang-auto">RequestFailedException: Request to 'https://api.github.com/repos/CraigChamberlain/moon-data/commits/' failed with: 422 UnprocessableEntity — {"message":"No commit found for SHA: ","documentation_url":"https://developer.github.com/v3/repos/commits/#get-a-single-commit"}
</code></pre>
<p>While I can’t speak to which HTTP client made this request, it may be worth highlighting that there’s a trailing slash, <code>/</code>,  at the end of this URL. Based on the error message, our API interpreted this request expecting a SHA after that slash.</p>
<p>It’s possible that you’re interested in <a href="https://developer.github.com/v3/repos/commits/#list-commits">listing commits for that repository</a>. If that’s the case, here’s an example <a href="https://curl.haxx.se/"><code>curl</code> request</a> for doing so:</p>
<pre><code class="lang-auto">curl --verbose https://api.github.com/repos/CraigChamberlain/moon-data/commits
</code></pre>
<p>On the other hand, if you’re interested in <a href="https://developer.github.com/v3/repos/commits/#get-a-commit">getting a single commit</a> for that repository, you can make a <a href="https://tools.ietf.org/html/rfc7231#section-4.3.1"><code>GET</code> request</a> to this URL. Here’s another example <a href="https://curl.haxx.se/"><code>curl</code> request</a>:</p>
<pre><code class="lang-auto">curl --verbose https://api.github.com/repos/CraigChamberlain/moon-data/commits/f438b25473961cfd0d57375c1af0e749c26aa97d
</code></pre>

**Comment by CraigChamberlain** [comment]
<p>Yes looks like the trailing slash was the issue. Silly of me to have missed it.</p>
<p>Thanks</p>
<p>C</p>
